src/middlewared/middlewared/plugins/directoryservices_/cache.py

```diff
@@ -183,6 +185,26 @@ def idmap_online_check_wait_wbclient(self, job):
 
         raise CallError('Timed out while waiting for domain to come online')
 
+    def idmap_online_check_wait_sssd(self, job):
+        """
+        SSSD reports a domain as online before it will _actually_ return results
+        for NSS queries. Since we know that enumeration is enabled if this is called
+        then we can use getpwent call to determine whether the domain is in a state
+        where we can actually fill our caches.
+        """
+        waited = 0
+        while waited <= 60:
+            for pwd in iterpw(module=NssModule.SSS.name):
+                # We have an entry provided by sssd
+                return
+
+            job.set_progress(10, 'Waiting for domain to come online')
```

Comment:
Maybe only log every 10 seconds so we don't pollute logs with 60 duplicate log messages?